packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -506,34 +520,55 @@ export function makeVatKeeper(
    * Store a snapshot, if given a snapStore.
    *
    * @param {VatManager} manager
-   * @returns {Promise<boolean>}
+   * @returns {Promise<void>}
    */
   async function saveSnapshot(manager) {
     if (!snapStore || !manager.makeSnapshot) {
-      return false;
+      return;
     }
 
+    // tell the manager to save a heap snapshot to the snapStore
     const endPosition = getTranscriptEndPosition();
     const info = await manager.makeSnapshot(endPosition, snapStore);
-    transcriptStore.rolloverSpan(vatID);
+
     const {
-      hash,
+      hash: snapshotID,
       uncompressedSize,
       rawSaveSeconds,
       compressedSize,
       compressSeconds,
     } = info;
+
+    // push a save-snapshot transcript entry
+    addToTranscript({
+      d: /** @type {TDSaveSnapshot} */ ['save-snapshot'],
+      sc: [],
+      r: { status: 'ok', snapshotID },
+    });
```

Comment:
Wondering if there should be functions to construct the various special transcript entries, rather than composing the raw structures inline.

Reply:
Maybe, but there's only one place it gets used, so there's no factoring argument in favor of a helper function.

Reply:
Yes, but there are several of these, each with the same complicated structure unfolded but with just minor variations.

Reply:
alright, you convinced me

Reply:
ugh, factoring them out into helper functions caused types to break